From bb0324e39824fc0de14fd4b9ad539849ff27c619 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Thu, 28 Nov 2019 12:46:12 +0000
Subject: [PATCH] abuild: fix applying patches from https

fix patches such as https://dev.alpinelinux.org/archive/domoticz/openzwave-1.6.patch

previously it would only work with the filenamename.patch::$url syntax
---
 abuild.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/abuild.in b/abuild.in
index 141cc54..5654d8f 100644
--- a/abuild.in
+++ b/abuild.in
@@ -682,7 +682,7 @@ default_prepare() {
 		case ${i%::*} in
 			*.patch)
 				msg "${i%::*}"
-				patch ${patch_args:--p1} -i "$srcdir/${i%::*}" || return 1
+				patch ${patch_args:--p1} -i "$srcdir/$(filename_from_uri $i)" || return 1
 				;;
 		esac
 	done
-- 
2.24.0

